osu.Game/Overlays/MusicController.cs

```diff
@@ -21,7 +24,7 @@ namespace osu.Game.Overlays
     /// <summary>
     /// Handles playback of the global music track.
     /// </summary>
-    public class MusicController : Component, IKeyBindingHandler<GlobalAction>
+    public class MusicController : CompositeDrawable, IKeyBindingHandler<GlobalAction>
```

Comment:
In general I'd be in favour of moving this out of the `osu.Game.Overlays` namespace out to `osu.Game.Audio`, but I wouldn't necessarily muddy this PR with that change as it's hard enough to read through as-is.

Reply:
Let's keep that for a separate PR.